Mathlib/RingTheory/DedekindDomain/Basic.lean

```diff
@@ -90,11 +98,35 @@ theorem DimensionLEOne.eq_bot_of_lt [Ring.DimensionLEOne R] (p P : Ideal R) [p.I
 
 end Ring
 
+/-- A Dedekind ring is a commutative ring that is Noetherian, integrally closed, and
+has Krull dimension at most one.
+
+This is exactly `IsDedekindDomain` minus the `IsDomain` hypothesis.
+
+The integral closure condition is independent of the choice of field of fractions:
+use `isDedekindRing_iff` to prove `IsDedekindRing` for a given `fraction_map`.
+-/
+class IsDedekindRing
+  extends IsNoetherian A A, DimensionLEOne A, IsIntegrallyClosed A : Prop
```

Comment:
Does this work
```suggestion
  extends IsNoetherianRing A, DimensionLEOne A, IsIntegrallyClosed A : Prop
```

Reply:
(It did not: `IsNoetherianRing` is not a structure.)